ui-particles-angular/projects/ui-policy-studio-angular/src/lib/gio-policy-studio.component.ts

```diff
@@ -0,0 +1,212 @@
+    private readonly location: Location,
+    @Inject('FlowService') private readonly flowService: FlowServiceAbstract,
+    @Inject('PolicyService') private readonly policyService: PolicyServiceAbstract,
+    @Inject('ResourceService') private readonly resourceService: ResourceServiceAbstract,
+    @Inject('SpelService') private readonly spelService: SpelServiceAbstract,
+  ) {}
+
+  ngOnInit(): void {
+    this.flowService
+      .getConfigurationSchemaForm()
+      .pipe(
+        tap(configurationSchema => {
+          this.configurationSchema = configurationSchema;
+        }),
+      )
+      .subscribe();
+
+    const { tabId, flowsIds } = this.parseUrl();
+
+    this.tabId = tabId;
+    this.selectedFlowsIds = JSON.stringify(flowsIds);
+  }
+
+  public onTabChanged(tabId: string): void {
+    this.updateUrl({ ...this.parseUrl(), tabId });
+  }
+
+  public onFlowSelectionChanged({ flows }: { flows: string[] }): void {
+    this.updateUrl({ ...this.parseUrl(), flowsIds: flows });
+  }
+
+  public fetchPolicyDocumentation({ policy }: { policy: { id: string; icon: string } }): void {
+    this.policyService
+      .getDocumentation(policy.id)
+      .pipe(tap(documentation => (this.policyDocumentation = { id: policy.id, image: policy.icon, content: documentation })))
+      .subscribe();
+  }
+
+  public fetchResourceDocumentation({
+    resourceType,
+    target,
+  }: {
+    resourceType: { id: string; icon: string };
+    target: { documentation: any };
+  }): void {
+    this.resourceService
+      .getDocumentation(resourceType.id)
+      .pipe(
+        tap(documentation => (target.documentation = { image: resourceType.icon, content: documentation })),
+        catchError(() => {
+          target.documentation = null;
+          return EMPTY;
+        }),
+      )
+      .subscribe();
+  }
+
+  public fetchSpelGrammar({ currentTarget }: { currentTarget: { grammar: Grammar } }): void {
+    this.spelService
+      .getGrammar()
+      .pipe(tap(grammar => (currentTarget.grammar = grammar)))
+      .subscribe();
+  }
+
+  private parseUrl(): UrlParams {
+    // TODO: Improve this with Angular Router
+    // Hack to add the tab as Fragment part of the URL
+    const [path, tabId] = this.location.path(true).split(/#(\w*)$/);
+
+    const [basePath, ...flowsIds] = path.split('flows');
+
+    const cleanedPath = basePath.replace('?', '');
+    const cleanedFlows = (flowsIds ?? []).map(flow => flow.replace('=', ''));
+
+    return {
+      path: cleanedPath,
+      tabId: tabId ?? '',
+      flowsIds: cleanedFlows,
+    };
+  }
+
+  private updateUrl({ path, tabId, flowsIds }: UrlParams): void {
+    // TODO: Improve this with Angular Router
+    // Hack to add the tab as Fragment part of the URL
+    const flowsQueryParams = (flowsIds ?? []).map(value => `flows=${value}`).join('&');
+
+    const queryParams = flowsQueryParams.length > 0 ? `?${flowsQueryParams}` : '';
+
+    this.location.go(`${path}${queryParams}#${tabId}`);
+  }
```

Comment:
since those two methods are hacks, maybe it worth to make them 'public' and fully unit test them ?

Reply:
for the moment I just move the code of apim here.

we can do some refactoring later.  we'll edit this part in the future I think to have angular routing